sql/hive/src/main/scala/org/apache/spark/sql/hive/orc/OrcFileFormat.scala

```diff
@@ -196,6 +198,11 @@ private[orc] class OrcSerializer(dataSchema: StructType, conf: Configuration)
 
   private[this] val cachedOrcStruct = structOI.create().asInstanceOf[OrcStruct]
 
+  private[this] val wrappers = dataSchema.zip(structOI.getAllStructFieldRefs().asScala.toSeq)
+      .map { case (f, i) =>
+    wrapperFor(i.getFieldObjectInspector, f.dataType)
+  }
```

Comment:
(my personal taste that might be worth considering.. 

```scala
private[this] val wrappers = dataSchema
  .zip(structOI.getAllStructFieldRefs.asScala.toSeq)
  .map { case (f, i) => wrapperFor(i.getFieldObjectInspector, f.dataType) }
```

or maybe

```scala
private[this] val wrappers = 
  structOI.getAllStructFieldRefs.asScala.toSeq.zip(dataSchema).map { case (ref, field) =>
    wrapperFor(ref.getFieldObjectInspector, field.dataType)
  }
```

which resembles https://github.com/apache/spark/blob/226d38840c8d3f40639715d755df6fb4fee2715f/sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveInspectors.scala#L364-L366
)